### ISSUE TITLE
SynRcvd should allow duplicate SYNs

### ISSUE THREAD

**Issue by rzezeski** [oxidecomputer/opte#186]
Today @bnaecker reported the following panic:

```
panic message: panicked at 'not yet implemented: bad packet: unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:192.168.1.145:63548, state: SYN_RCVD, flags: 0x2', /work/oxidecomputer/opte/opte/src/engine/port.
```

This is caused by an inbound SYN from `192.168.1.145` on his home network to the guest `172.30.0.5` when the TCP state machine was already in the `SyncRcvd` state. In that state you would typically expect an incoming `ACK` from the client (`192.168.1.145`) in order to finish establishing the connection. However, Ben also reported the following:

```
I didn't update the gateway MAC here, I was using the lab's which is def wrong for my home network.
```

In this case he is referring to the "external IP hack", which uses his home's network gateway to relay NAT'd packets between his home network and the OPTE guest. With an incorrect MAC address this would mean all frames sent out from the guest would be dropped by the switch connected to the host/server (running Omicron) -- no traffic would flow out. However, since OPTE sees the outbound guest traffic it knows the guest is in a SynRcvd state and waiting for the ACK. But since the client never saw the `SYN + ACK` from the guest it decides to retransmit its original `SYN` to the guest. This is totally normal TCP behavior and should be allowed by the OPTE TCP state machine.

**Comment by rzezeski** [comment]
Just to be sure I replicated this issue by providing an incorrect value for the `gateway_mac` in sled-agent's `config.toml`. With that I then tried to SSH from my laptop to a newly created guest and replicated the expected panic.

```
> ::status
debugging crash dump /var/crash/kalm/vmcore.41 (64-bit) from kalm
operating system: 5.11 helios-1.0.21180 (i86pc)
build version: remotes/ci-build/xde-0-gd365ee6a2b

image uuid: c403cf10-bb8f-4d71-af00-cc3d2dfdeb57
panic message: panicked at 'not yet implemented: bad packet: unexpected TCP segment dir: IN, flow: TCP:172.30.0.5:22:10.0.0.10:57561, state: SYN_RCVD, flags: 0x2', /home/rpz/oxidecomputer/opte/opte/src/engine/port.
dump content: kernel pages only
> 
> $C
fffffe0079a08800 vpanic()
fffffe0079a08880 rust_begin_unwind+0x8a()
fffffe0079a088c0 core::panicking::panic_fmt::h7d7971299bdd0756+0x2d()
fffffe0079a08990 opte::engine::port::Port::bad_packet_err::hbaf4f641a8504f55+0x1cc()
fffffe0079a08c10 opte::engine::port::Port::process::hecebe1beb9b8488a+0x3487()
fffffe0079a09740 xde_rx+0x4a4()
fffffe0079a09790 mac_promisc_dispatch_one+0xac(fffffe58c10e7ee8, fffffe592a2f4660, 0, 0)
fffffe0079a097f0 mac_promisc_client_dispatch+0xb4(fffffe59289a4a48, fffffe592a2f4660)
fffffe0079a09870 mac_rx_srs_drain+0x1d2(fffffe5899bc6900, 800)
fffffe0079a09900 mac_rx_srs_process+0x123(fffffe5896c88010, fffffe5899bc6900, fffffe592a2f4660, 0)
fffffe0079a09950 mac_rx_classify+0x7c(fffffe5896c88010, 0, fffffe592a2f4660)
fffffe0079a099c0 mac_rx_flow+0x58(fffffe5896c88010, 0, fffffe592a2f4660)
fffffe0079a09a20 mac_rx_common+0x1de(fffffe5896c88010, 0, fffffe592a2f4660)
fffffe0079a09a80 mac_rx+0xc6(fffffe5896c88010, 0, fffffe592a2f4660)
fffffe0079a09ac0 mac_rx_ring+0x1f(fffffe5896c88010, 0, fffffe592a2f4660, 0)
fffffe0079a09b20 e1000g_intr_work+0x264(fffffe58901b4000, 80000081)
fffffe0079a09b50 e1000g_intr_pciexpress+0xa3(fffffe58901b4000, 0)
fffffe0079a09bd0 apix_dispatch_pending_autovect+0xef(6)
fffffe0079a09c00 apix_dispatch_pending_hardint+0x34(0, 0)
fffffe00799c19f0 switch_sp_and_call+0x15()
fffffe00799c1a60 apix_do_interrupt+0x42d(fffffe00799c1a70, 10)
fffffe00799c1a70 _interrupt+0xc3()
fffffe00799c1ba0 fakesoftint_return()
fffffe00799c1be0 _resume_from_idle+0x12b()
fffffe00799c1c00 idle+0x10c()
fffffe00799c1c10 thread_start+0xb()
```

**Comment by rzezeski** [comment]
With the patch applied I can see the `SYN` being retransmitted by the client while the TCP flow is in the `SynRcvd` state.

```
rpz@kalm:~/opte-dtrace$ pfexec ./opte-trace opte-tcp-flow-state.d 
PORT             CURR         NEW          FLOW
opte0            LISTEN       SYN_RCVD     TCP,172.30.0.5:22,10.0.0.10:57587

rpz@kalm:~$ pfexec snoop -r -d net0 '(src 10.0.0.226) or (dst 10.0.0.226)'
Using device net0 (promiscuous mode)
   10.0.0.10 -> 10.0.0.226   TCP D=22 S=57587 Syn Seq=3833175229 Len=0 Win=65535 Options=<mss 1460,nop,wscale 6,nop,nop,tstamp 3205155057 0,sackOK,eol>
  10.0.0.226 -> 10.0.0.10    TCP D=57587 S=22 Syn Ack=3833175230 Seq=2152564421 Len=0 Win=65160 Options=<mss 1460,sackOK,tstamp 199534699 3205155057,nop,wscale 7>
   10.0.0.10 -> 10.0.0.226   TCP D=22 S=57587 Syn Seq=3833175229 Len=0 Win=65535 Options=<mss 1460,nop,wscale 6,nop,nop,tstamp 3205156058 0,sackOK,eol>

rpz@kalm:~/oxidecomputer/opte/opteadm$ pfexec opteadm dump-tcp-flows -p opte0
TCP:172.30.0.5:22:10.0.0.10:57587 FlowEntryDump { hits: 0, state_summary: "TCP:10.0.0.10:57587:10.0.0.226:22 SYN_RCVD Some(2152564421) Some(3833175230) Some(3833175229) None" }
```

**Comment by rzezeski** [comment]
Addressed in eb9e0c687e3c072dbc7d4782475f4ba5a6f50258.
